### ISSUE TITLE
Switch to crossref plus API

### ISSUE THREAD

**Issue by erkannt** [sciety/sciety#310]
- [x] Add the token for Plus access as a request header in the following format `Crossref-Plus-API-Token: Bearer FullTokenHere`. For those with access the token is here: https://github.com/elifesciences/it-admin/blob/master/crossref-plus.md
  - [x] add to `scripts`
  - [x] add to `fetch-crossref-article`
  - [x] add check that we are loading from `api.crossref.org`
  - [x] decide whether to inject key into `getXml` or hold off
  - [x] ~consider stripping out article cache~ keep it as it provides de-duplication of concurrent requests, protects against downtime
  - [x] consider renaming `getXml` to be Crossref specific
  - [x] provide token to staging and prod via GH secrets
  - [x] remove Disqus environment variable
  - [x] consider pushing DOI to `getXmlFromCrossrefRestApi`
     - [x] change type of `GetXml` and conform
     - [x] ~can we move `catch` on the other side of the port by replacing the exception with a `Result`?~
     - [x] consistency in where the headers come from; `Accept` vs `User-Agent` vs token
  - [x] add a request duration log once each request has finished in `fetch-crossref-article`
- [x] display Crossref's DOIs according to their specification https://doi.org/10.13003/5jchdy as it's a part of the agreement with Crossref

**Comment by BlueReZZ** [comment]
Some additional information from the registration email, duplicated from the secret page above:

The token should be used as follows:

`Crossref-Plus-API-Token: Bearer FullTokenHere`

It may be used for either or both OAI-PMH and the REST API. If you do decide to use OAI-PMH, please point here: https://oai.crossref.org/oai

For the REST API, even though we have a token, they still recommend identifying ourselves with an email address as a ‘polite’ user, as described here, along with other API documentation: https://github.com/CrossRef/rest-api-doc#etiquette That will help them get in touch quickly should something really wrong need to be sorted out.

A few other things to note from Crossref:

- You can keep an eye on our current uptime and sign up for alerts with our public status page: http://status.crossref.org/
- When using the REST API, you’ll need to use query.bibliographic instead of query.title
- Of note should be the instructions for the monthly full-file "snapshots": https://www.crossref.org/help/metadata-snapshots/
- We’ve created a new "badge" to reflect your relationship with Crossref, for use on websites and and in other communications, if you’d like (let me know if you need it in other formats). You’ll find a ‘Metadata User’ badge here, along with some information on using the file: https://www.crossref.org/brand/member-badges/

**Comment by thewilkybarkid** [comment]
At runtime we currently request the DOI directly with an `Accept` header of Crossref's type rather than using their API:

```sh
curl https://doi.org/[doi] -H "Accept: application/vnd.crossref.unixref+xml"
```

This redirects to `https://data.crossref.org/[doi]` which returns the article metadata in the `application/vnd.crossref.unixref+xml` format. This appears _not_ to make use of the `Crossref-Plus-API-Token` header.

Using their API instead:

```sh
curl -v https://api.crossref.org/works/[doi]
```

a [different format is returned (JSON)](https://github.com/CrossRef/rest-api-doc/blob/master/api_format.md). This does include the abstract in JATS (rather than Crossref XML), but the title appears to be plain text (which would make fixing #324 impossible). 

There was a decision to [add a `title-xml` property over 2 years ago](https://github.com/CrossRef/rest-api-doc/issues/112#issuecomment-376221650), but appears not to have been done.

We've happened to stumble across https://github.com/CrossRef/rest-api-doc/issues/103#issuecomment-172317784, which mentions an _undocumented_ API for returning the same type as `data.crossref.org`:

```sh
curl https://api.crossref.org/works/[doi]/transform -H "Accept: application/vnd.crossref.unixref+xml"
```

**Comment by giorgiosironi** [comment]
1. Use the undocumented API as a drop-in replacement
2. @BlueReZZ to ask Crossref clarifications about state of support and/or possible replacements to get a `title` with a structured format (e.g. XML).

**Comment by gbilder** [comment]
Meta- we have been slowly switching from using Github tickets and Jira to just using Gitlab issues.

You are correct to be suspicious of using the undocumented transform, we outline some of the problems you may encounter here:

https://gitlab.com/crossref/issues/-/issues/903

Unfortunately, as you surmised, there is no real alternative because the JSON in the REST API strips markup. You are also correct that this is a long-standing bug. The reason for this is that we've had a code freeze in place as we migrate the REST API backend to use Elasticsearch instead of Solr and to be hosted on AWS instead of in our own data centre. We think the revamped API will launch in Q1 2021 and *then* we can start working on fixing some of these long-outstanding bugs. The title bug is an important one that effects many of our users.

Having said that, it also looks like the ticket was not migrated to Gitlab when we moved from Github/JIRA-  so I have opened it again in Gitlab.. Feel free to comment, etc. Comments will help us prioritise.

https://gitlab.com/crossref/issues/-/issues/1001

In short-  I think you have made an  pragmatic choice to use the transforms- but would recommend that you switch to using the JSON and your own CSL formatting as soon as we get titles properly represented in the JSON.
